tests/components/vesync/test_init.py

```diff
@@ -0,0 +1,198 @@
+    ) as setup_mock:
+        assert await async_setup_entry(hass, config_entry)
+        # Assert platforms loaded
+        await hass.async_block_till_done()
+        assert setups_mock.call_count == 1
+        assert setups_mock.call_args.args[0] == config_entry
+        assert setups_mock.call_args.args[1] == []
+        assert setup_mock.call_count == 0
+    assert manager.login.call_count == 1
+    assert hass.data[DOMAIN][VS_MANAGER] == manager
+    assert not hass.data[DOMAIN][VS_SWITCHES]
+    assert not hass.data[DOMAIN][VS_FANS]
+    assert not hass.data[DOMAIN][VS_LIGHTS]
+    assert not hass.data[DOMAIN][VS_SENSORS]
+
+
+async def test_async_setup_entry__loads_fans(
+    hass: HomeAssistant, config_entry: ConfigEntry, manager: VeSync, fan
+):
+    """Test setup connects to vesync and loads fan platform."""
+    fans = [fan]
+    manager.fans = fans
+    manager._dev_list = {
+        "fans": fans,
+    }
+
+    with patch.object(
+        hass.config_entries, "async_forward_entry_setups"
+    ) as setups_mock, patch.object(
+        hass.config_entries, "async_forward_entry_setup"
+    ) as setup_mock:
+        assert await async_setup_entry(hass, config_entry)
+        # Assert platforms loaded
+        await hass.async_block_till_done()
+        assert setups_mock.call_count == 1
+        assert setups_mock.call_args.args[0] == config_entry
+        assert setups_mock.call_args.args[1] == [Platform.FAN, Platform.SENSOR]
+        assert setup_mock.call_count == 0
+    assert manager.login.call_count == 1
+    assert hass.data[DOMAIN][VS_MANAGER] == manager
+    assert not hass.data[DOMAIN][VS_SWITCHES]
+    assert hass.data[DOMAIN][VS_FANS] == [fan]
+    assert not hass.data[DOMAIN][VS_LIGHTS]
+    assert hass.data[DOMAIN][VS_SENSORS] == [fan]
+
+
+async def test_async_setup_entry__loads_bulbs(
+    hass: HomeAssistant, config_entry: ConfigEntry, manager: VeSync, bulb
+):
+    """Test setup connects to vesync and loads light platform."""
+    bulbs = [bulb]
+    manager.bulbs = bulbs
+    manager._dev_list = {
+        "bulbs": bulbs,
+    }
+
+    with patch.object(
+        hass.config_entries, "async_forward_entry_setups"
+    ) as setups_mock, patch.object(
+        hass.config_entries, "async_forward_entry_setup"
+    ) as setup_mock:
+        assert await async_setup_entry(hass, config_entry)
+        # Assert platforms loaded
+        await hass.async_block_till_done()
+        assert setups_mock.call_count == 1
+        assert setups_mock.call_args.args[0] == config_entry
+        assert setups_mock.call_args.args[1] == [Platform.LIGHT]
+        assert setup_mock.call_count == 0
+    assert manager.login.call_count == 1
+    assert hass.data[DOMAIN][VS_MANAGER] == manager
+    assert not hass.data[DOMAIN][VS_SWITCHES]
+    assert not hass.data[DOMAIN][VS_FANS]
+    assert hass.data[DOMAIN][VS_LIGHTS] == [bulb]
+    assert not hass.data[DOMAIN][VS_SENSORS]
+
+
+async def test_async_setup_entry__loads_switches(
```

Comment:
it's a bit overkill to copy paste this test and test single platforms. If you want to take this approach, use `@pytest.mark.parametrize`

Reply:
I've been doing working on another vesync PR (https://github.com/home-assistant/core/pull/84025) that has another batch of test improvements.  I'd like to incorporate that test bed before trying to fix this, if possible.  I can remove these test cases, if you would rather not have the extra test methods.

Reply:
Let's keep them out of this PR then.

Reply:
Removed the extra test cases.  Will add them back in for coverage with an upcoming PR